static/monitoring.js

```diff
@@ -109,6 +112,51 @@ function appendLine(element, text) {
     element.appendChild(document.createElement("br"));
 }
 
+function fetchHistory(parent, token, fingerprint, offsetId, directive, page=0) {
+    parent.innerHTML = '';
+    if (page > 0) {
+        parent.appendChild(create("button", "Last 20", "page_history_prev"));
+    }
+    fetch("/api/attestation_history.json", {method: "POST", body: {token, fingerprint,
+        offsetId, directive}, credentials: "same-origin"}).then(response => {
+        if (!response.ok) {
+            return Promise.reject();
+        }
+        return response.json();
+    }).then(attestations => {
+        for (const attestation of attestations) {
+            history.appendChild(create("h4", new Date(attestation.time)));
+
+            const p = history.appendChild(document.createElement("p"));
+            const result = attestation.strong ?
+                "Successfully performed strong paired verification and identity confirmation." :
+                "Successfully performed basic initial verification and pairing.";
+            p.appendChild(create("strong", result));
+
+            history.appendChild(create("h5", "Verified device information (constants omitted):"));
+            history.appendChild(create("p", attestation.teeEnforced));
+            history.appendChild(create("h5", "Information provided by the verified OS:"));
+            history.appendChild(create("p", attestation.osEnforced));
+        }
+        if (attestations.length === 20) {
+            parent.appendChild(create("button", "Next 20", "page_history_next"));
+            parent.offsetId = attestations[attestations.length - 1].offsetId;
```

Comment:
You need to use data prefixes. You can use the `dataset` syntactic sugar for this. It's not well-defined to set arbitrary attributes on DOM nodes.